File: yogi-dotnet/test/TimeTest.cs
using System;
using Xunit;

namespace test
{
    public class TimeTest : TestCase
    {
        [Fact]
        public void CurrentTime()
        {
            var start = Yogi.Timestamp.Now;
            var t = Yogi.CurrentTime;
            Assert.IsType<Yogi.Timestamp>(t);
            Assert.InRange(t, start, Yogi.Timestamp.Now);
        }
    }
}

File: yogi-cpp/test/time_test.cc
#include "common.h"

#include <chrono>
using namespace std::chrono_literals;

TEST(TimeTest, GetCurrentTime) {
  auto t = yogi::GetCurrentTime();
  auto now = std::chrono::system_clock::now();
  EXPECT_LT((now - t) + 1s, 2s);
}
